DOCS-3930 Moving Around the Synthetic Testing Tunnel (#15316)
DataDog/documentation: content/en/continuous_testing/troubleshooting.md

---
title: Continuous Testing and CI/CD Troubleshooting
kind: documentation
description: Learn about Continuous Testing and CI/CD concepts and troubleshoot common errors.
aliases:
  - /synthetics/cicd_integrations/troubleshooting
further_reading:
- link: "/synthetics/cicd_integrations/configuration"
  tag: "Documentation"
  text: "Learn how to configure Continuous Testing and CI/CD"
---

## Overview

> This page provides information to help you troubleshoot issues with Continuous Testing and CI/CD. If you need additional help, contact [Datadog support][1].

Author: Asking Beth!

---

> title: Synthetics and CI/CD Troubleshooting

Reviewer: ```suggestion
title: Continuous Testing and CI/CD Troubleshooting
```

kind: documentation

> description: Learn about Synthetics and CI/CD concepts and troubleshoot common errors.

Reviewer: ```suggestion
description: Learn about Continuous Testing and CI/CD concepts and troubleshoot common errors.
```

Execution rule
: An [execution rule][4] defines the impact of a test failure on a CI pipeline from most to least impactful: `skipped`, `non_blocking`, and `blocking`. These options are weighted and defaults to the most impactful. If your test is configured as `skipped` in the UI and `blocking` in the configuration file, it is skipped during the test run. </br><br> You can set the execution rule in your tests' properties, global configuration file, or an individual test's override file. 

> ## CI Results Explorer

Author: ```suggestion
## Explorer
```

### CI metadata does not appear

> Check whether you are using API endpoints to trigger your CI/CD test runs. In order to populate the CI Results Explorer with CI metadata, you must use the [NPM package][5].

Reviewer: ```suggestion
Check whether you are using API endpoints to trigger your CI/CD test runs. In order to populate the CI Results Explorer with CI metadata, you must use one of our integrations under [CI/CD integrations][3].
```

Thinking we should point to the GitHub Action, Circle CI orb, Azure DevOps Extension and Jenkins integration, before deferring them to the NPM package.
